Core/Object Arts/Dolphin/Base/CompiledMethod.cls

```diff
@@ -77,12 +77,9 @@ hasChanged
 icon
 	"Answers an Icon that can be used to represent this object."
 
-	self isDeprecated ifTrue: [^self class deprecatedIcon].
-
-	^self isPrivate
-		ifTrue: [self class privateIcon]
-		ifFalse: [self class publicIcon]
-!
+	^(Smalltalk at: #SmalltalkSystem ifPresent: [:a | a current])
```

Comment:
Since this super-sends when the development system is not present, a simpler solution would be to make it a loose method in the devsystem package, and then it can just be `^SmalltalkSystem current iconForMethod: self`